gapic/src/main/com/google/gapid/widgets/LoadableImage.java

```diff
@@ -105,13 +102,10 @@ protected void onUiThreadError(Void error) {
   }
 
   public LoadableImage unload() {
-    if (--loadCount != 0) {
+    if (--loadCount != 0 || state != State.LOADING) {
```

Comment:
The reason I didn't merge these if statements together is it's now less obvious that you're pre-decrementing. Given that swapping the order of these two conditionals can introduce bugs, I'd strongly recommend doing the decrement on a line above.